### PR TITLE
Fixes #2056 - Needs tests!

### LINKED ISSUE #2056: [enhancement] Set RTC when receiving GPS timestamp
The Taranis RTC has the habit to drift badly (at least mine). Wouldn't it be possible to correct it when receiving GPS timestamps via telemetry?

### PR DESCRIPTION
**Pull request by bsongis**

### CODE REVIEW

File: radio/src/telemetry/telemetry.cpp [hunk 1]
```diff
@@ -53,12 +53,28 @@ void TelemetryItem::setValue(const TelemetrySensor & sensor, int32_t newVal, uin
...
       datetime.sec = (uint16_t) ((data & 0x0000ff00) >> 8);
       datetime.timestate = 1;
+      if (g_eeGeneral.adjustRTC) {
+        struct gtm t;
+        gettime(&t);
+        t.tm_hour = datetime.hour;
+        t.tm_min = datetime.min;
+        t.tm_sec = datetime.sec;
+        rtcSetTime(&t);
+      }
```
Comment (projectkk2glider) [inline radio/src/telemetry/telemetry.cpp]: Due to the telemetry stream delay, I think we will have a significant jitter in RTC time if we set it every time the new GPS time-stamp comes in. I propose we only set the RTC time if the RTC and GPS clock differ more than X seconds (I propose X of around 20s).

I also think you forgot to set:
```c
g_rtcTime = gmktime(&t); // update local timestamp and get wday calculated
```

Comment (bsongis) [inline radio/src/telemetry/telemetry.cpp]: True, thanks!

**Comment by bsongis** [comment]
Reminder ... Companion part

**Comment by kilrah** [comment]
Works fine!
Should we maybe limit to doing only one auto-adjust per session (power cycle) to avoid potential jumps in log timestamps if the RTC drifts a lot? Not sure if it could be that bad of it it's never going to be a problem...

**Comment by kilrah** [comment]
There is still an issue with building the sky9x simu on that one at least on Win:

>3>     Creating library C:/opentx/c9x_build_win/firmwares/opentx/simulator/Release/opentx-sky9x-en-simulator.lib and object C:/opentx/c9x_build_win/firmwares/opentx/simulator/Release/opentx-sky9x-en-simulator.exp
3>opentxsimulator.obj : error LNK2019: unresolved external symbol "void __cdecl Sky9X::rtcSetTime(struct Sky9X::gtm *)" (?rtcSetTime@Sky9X@@YAXPAUgtm@1@@Z) referenced in function "public: void __thiscall Sky9X::TelemetryItem::setValue(struct Sky9X::TelemetrySensor const &,int,unsigned int,unsigned int)" (?setValue@TelemetryItem@Sky9X@@QAEXABUTelemetrySensor@2@HII@Z)
3>opentxsimulator.obj : error LNK2019: unresolved external symbol "long Sky9X::g_rtcTime" (?g_rtcTime@Sky9X@@3JA) referenced in function "public: void __thiscall Sky9X::TelemetryItem::setValue(struct Sky9X::TelemetrySensor const &,int,unsigned int,unsigned int)" (?setValue@TelemetryItem@Sky9X@@QAEXABUTelemetrySensor@2@HII@Z)
3>opentxsimulator.obj : error LNK2019: unresolved external symbol "long __cdecl Sky9X::gmktime(struct Sky9X::gtm *)" (?gmktime@Sky9X@@YAJPAUgtm@1@@Z) referenced in function "public: void __thiscall Sky9X::TelemetryItem::setValue(struct Sky9X::TelemetrySensor const &,int,unsigned int,unsigned int)" (?setValue@TelemetryItem@Sky9X@@QAEXABUTelemetrySensor@2@HII@Z)
3>opentxsimulator.obj : error LNK2019: unresolved external symbol "void __cdecl Sky9X::gettime(struct Sky9X::gtm *)" (?gettime@Sky9X@@YAXPAUgtm@1@@Z) referenced in function "public: void __thiscall Sky9X::TelemetryItem::setValue(struct Sky9X::TelemetrySensor const &,int,unsigned int,unsigned int)" (?setValue@TelemetryItem@Sky9X@@QAEXABUTelemetrySensor@2@HII@Z)
3>C:\opentx\c9x_build_win\Release\opentx-sky9x-en-simulator.dll : fatal error LNK1120: 4 unresolved externals

The rest is OK, so that can be merged when the linking issue is fixed.

**Pull request merged**